src/prefect/blocks/core.py

```diff
@@ -660,7 +690,22 @@ class Custom(Block):
                 f"Unable to find block document named {block_document_name} for block type {block_type_slug}"
             ) from e
 
-        return cls._from_block_document(block_document)
+        try:
+            return cls._from_block_document(block_document)
+        except ValidationError as e:
+            if skip_validation:
+                missing_fields = tuple(err["loc"][0] for err in e.errors())
+                missing_block_data = {field: None for field in missing_fields}
+                warnings.warn(
+                    f"Could not fully load {block_document_name!r} of block type {cls.__name__!r} - "
+                    "this is likely because one or more required fields were added to the schema "
+                    f"for {cls.__name__!r} that did not exist on the class when this block was last saved. "
+                    f"Please specify values for new field(s): {listrepr(missing_fields)}, then "
+                    f'run `{cls.__name__}.save("{block_document_name}", overwrite=True)`, and '
+                    "load this block again before attempting to use it."
+                )
+                return cls.construct(**block_document.data, **missing_block_data)
```

Comment:
What happens when fields were removed from the block type?

Reply:
This should be covered without `skip_validation`, so if we had
```python

class Foo(Block):
   x: int
   y: int

Foo(x=1, y=2).save("xy")

# and then

class Foo(Block):
   x: int

Foo.load("xy") # yields Foo(x=1) without skipping validation

Reply:
Thanks for confirming. Maybe you can add this as a comment?

Reply:
will do - also adding a test now to verify this